pkg/extensions/tbtc/tbtc.go

```diff
@@ -761,23 +790,58 @@ func (t *tbtc) watchKeepClosed(
 }
 
 func (t *tbtc) shouldMonitorDeposit(depositAddress string) bool {
```

Comment:
What do you think about returning `(bool, int)` from this function and having `timeoutFn` accept `memberIndex int` instead of `depositAddress string`? We can spare two Ethereum client calls this way.

Reply:
What in the case when we get the result from the cache?

Reply:
Ah yes. OK, it's more complex. Let's leave it as is, we can always revisit later.